File: 200/228 - Summary Ranges.java
import java.util.ArrayList;

public class Solution {
    public List<String> summaryRanges(int[] nums) {
        int from, to, i, c;
        List<String> ret = new ArrayList<String>();
        
        if (nums.length == 0)
            return ret;
        to = nums[0];
        from = to;
        
        for (i = 1; i < nums.length; ++i) {
            c = nums[i];
            if (c != to + 1) {
                ret.add(from == to ? Integer.toString(from) : from + "->" + to);
                from = c;
            }
            
            to = c;
        }
        
        if (to != nums[0] || ret.isEmpty())
            ret.add(from == to ? Integer.toString(from) : from + "->" + to);
        
        return ret;
    }
}

File: csharp/228.summary-ranges.cs
/*
 * @lc app=leetcode id=228 lang=csharp
 *
 * [228] Summary Ranges
 */

using System;

public class Solution {
    public IList<string> SummaryRanges(int[] nums) {
        if (nums.Length == 0) {
            return Array.Empty<string>();
        }
        var ret = new List<string>();
        int start = nums[0], end = start;
        for (int i = 1; i < nums.Length; ++i) {
            int num = nums[i];
            if (num == end + 1) {
                ++end;
            } else {
                ret.Add(start == end ? start.ToString() : $"{start}->{end}");
                start = num;
                end = num;
            }
        }
        ret.Add(start == end ? start.ToString() : $"{start}->{end}");
        return ret;
    }
}

